main.tf

```diff
@@ -7,6 +7,18 @@ provider "aws" {
   }
 }
 
+resource "random_integer" "externalid" {
```

Comment:
Bit preference, maybe use https://registry.terraform.io/providers/hashicorp/random/latest/docs/resources/uuid? @shoekstra you have an opinion here? 
limits should allow this: https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_iam-quotas.html
he external ID value that a third party uses to assume a role must have a minimum of 2 characters and a maximum of 1,224 characters. The value must be alphanumeric without white space. It can also include the following symbols: plus (+), equal (=), comma (,), period (.), at (@), colon (:), forward slash (/), and hyphen (-). For more information about the external ID, see [How to use an external ID when granting access to your AWS resources to a third party](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_create_for-user_externalid.html).

Reply:
Why would you switch? It's not a UUID, it has nothing to do with a UUID...